src/Admin/AdminConsole/Controllers/ProvidersController.cs

```diff
@@ -275,4 +277,64 @@ public async Task<IActionResult> CreateOrganization(Guid providerId, Organizatio
 
         return RedirectToAction("Edit", "Providers", new { id = providerId });
     }
+
+    [HttpPost]
+    [SelfHosted(NotSelfHostedOnly = true)]
+    [RequirePermission(Permission.Provider_Edit)]
+    public async Task<IActionResult> Delete(Guid id, string providerName)
+    {
+        var providerOrganizations = await _providerOrganizationRepository.GetManyDetailsByProviderAsync(id);
+
+        if (providerOrganizations.Count == 0)
+        {
+            var provider = await _providerRepository.GetByIdAsync(id);
+
+            if (provider is null)
+            {
+                return BadRequest("provider does not exist");
+            }
+
+            if (providerName.ToLower().Trim() != provider.Name.ToLower().Trim())
+            {
+                return BadRequest("Invalid provider name");
+            }
+
+            await _providerService.DeleteAsync(provider);
+        }
+
+        return NoContent();
```

Comment:
❓ In my opinion, responding with No Content here gives the impression that this operation was successful when in reality, it was invalid. How is this handled by the front-end?

Reply:
This has been refactored. but The front end is validating if the provider has a client and if only when there is no client that his method is called.